processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -358,58 +399,96 @@ public static Float convertObjectToFloat(@Nullable Object valObj)
   @Nullable
   public static Float convertObjectToFloat(@Nullable Object valObj, boolean reportParseExceptions)
   {
-    if (valObj == null) {
-      return null;
-    }
+    return convertObjectToFloat(valObj, reportParseExceptions, null);
+  }
 
-    if (valObj instanceof Float) {
-      return (Float) valObj;
-    } else if (valObj instanceof Number) {
-      return ((Number) valObj).floatValue();
-    } else if (valObj instanceof String) {
-      Float ret = Floats.tryParse((String) valObj);
-      if (reportParseExceptions && ret == null) {
-        throw new ParseException((String) valObj, "could not convert value [%s] to float", valObj);
+  @Nullable
+  public static Float convertObjectToFloat(@Nullable Object valObj, @Nullable String fieldName)
```

Comment:
Question on the `@Nullable` annotation: this new overloaded function appears to be called only from `SqlResults` similar to `convertObjectToLong()` where the fieldName doesn't have a `@Nullable` annotation. I don't think `fieldName` can be nullable in this context, right? If that's correct, please remove the annotation here; otherwise, update `convertObjectToLong`'s signature above to reflect that